src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStep.java

```diff
@@ -86,6 +86,9 @@ public DescriptorImpl() {
             super(BuildTriggerStepExecution.class);
         }
 
+        // Note: This is necessary because the JSON format of the parameters produced by config.jelly when
+        // using the snippet generator does not match what would be neccessary for databinding to work automatically.
+        // For non-snippet generator use, this is unnecessary.
         @Override public Step newInstance(StaplerRequest req, JSONObject formData) throws FormException {
```

Comment:
BTW you could update the `structs` dep and use `CustomDescribableModel`.

Reply:
Is there something that ties `Descriptor.newInstance(StaplerRequest, JSONObject)` to `DescribableModel`, or do you mean implement `CustomDescribableModel` and replace the body here with something like the following?

```
   return DescribableModel.of(BuildTriggerStep.class).instantiate(convertJsonToMap(formData))
```

Also, unless it is safe to assume that data binding for `ParameterDefinition`s via `DescribableModel` will operate identically to `ParameterDefinition.createValue(StaplerRequest, JSONObject)` we'd still have to use that method in `customInstantiate`, which would be a bit awkward.

Seems easiest just to leave it for now.

Reply:
Sorry, forgot this was the overload with `JSONObject`. Ignore.